rego/rego.go

```diff
@@ -1689,7 +1689,14 @@ func (r *Rego) prepare(ctx context.Context, qType queryType, extras []extraStage
 }
 
 func (r *Rego) parseModules(ctx context.Context, txn storage.Transaction, m metrics.Metrics) error {
-	if len(r.modules) == 0 {
+	ids, err := r.store.ListPolicies(ctx, txn)
```

Comment:
Is this operation safe to put ahead of the metrics timers? :thinking: Previously, this operation ran only *after* the timers had been started, and it looks like it hits the store with a transaction to go and iter through all the available policies.

I guess my worry is that if the store is slow to do the transaction for iterating over the policies, it might skew measurements slightly? @srenatus or @ashutosh-narkar might have better knowledge over whether it will matter or not. :sweat_smile:

Reply:
Hmm, that's a thought. My thinking was that we wanted to time the _parsing_ part of this function only, but happy to be corrected.

Reply:
I think I'm with Charlie here. If we had been including the store retrieval times in the parse metric before; and don't do it now, it's for the better. Less noise in the metrics. I also don't believe anyone really depends on those metrics in a way that the change would break stuff.